app/models/openscap_result.rb

```diff
@@ -0,0 +1,76 @@
+class OpenscapResult < ApplicationRecord
+  belongs_to :container_image
+  has_one    :binary_blob,           :dependent => :destroy, :autosave => true, :as => :resource
+  has_many   :openscap_rule_results, :dependent => :destroy, :autosave => true
+
+  def self.openscap_available?
+    # needed only for travis
+    require 'openscap'
+    require 'openscap/ds/arf'
+    require 'openscap/xccdf/benchmark'
+    true
+  rescue LoadError
+    false
+  end
+
+  def raw=(value)
+    raise "Cannot invoke raw more than once." unless binary_blob.nil?
```

Comment:
I think what you want here is a couple of things:

1. @durandom's suggestion about `validates :binary_blob, presence: true`
2. possibly move the creation of the child `openscap_rule_results` to a `before_save`
3. make `raw` readonly so it can only be set when the `openscap_result` object is new

That way, it doesn't matter how many times `raw` is called on a new `openscap_result` object, it only gets exploded once during save, and once it's save it's immutable.

Does that sound about right?  Or, am I missing the point?